controllers/feeds/getOneFeedChat.js

```diff
@@ -24,14 +37,30 @@ const getReaction = (req, latest_reactions, constantActor) => {
   const upvotes = [];
   const downvotes = [];
   if (Array.isArray(latest_reactions.comment)) {
-    latest_reactions.comment.map((comment) => constructData(req, comments, comment, constantActor));
+    latest_reactions.comment.map((comment) =>
+      constructData({selfUserId: req?.userId, data: comments, datum: comment, constantActor})
+    );
   }
   if (Array.isArray(latest_reactions.upvotes)) {
-    latest_reactions.upvotes.map((upvote) => constructData(req, upvotes, upvote, constantActor));
+    latest_reactions.upvotes.map((upvote) =>
+      constructData({
+        selfUserId: req?.userId,
+        data: upvotes,
+        datum: upvote,
+        constantActor,
+        shouldIncludeActorData: false
+      })
+    );
   }
   if (Array.isArray(latest_reactions.downvotes)) {
     latest_reactions.downvotes.map((downvote) =>
-      constructData(req, downvotes, downvote, constantActor)
+      constructData({
+        selfUserId: req?.userId,
+        data: downvotes,
+        datum: downvote,
+        constantActor,
+        shouldIncludeActorData: false
+      })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are using the `map` function to iterate over arrays and perform side effects. While this works, it's not the best practice. The `map` function is intended to create a new array based on the original one, not to perform side effects. For side effects, use `forEach` instead. 

```diff
- latest_reactions.comment.map((comment) =>
+ latest_reactions.comment.forEach((comment) =>
  
- latest_reactions.upvotes.map((upvote) =>
+ latest_reactions.upvotes.forEach((upvote) =>

- latest_reactions.downvotes.map((downvote) =>
+ latest_reactions.downvotes.forEach((downvote) =>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->